core/src/test/scala/integration/kafka/api/SaslSslAdminClientIntegrationTest.scala

```diff
@@ -119,4 +150,126 @@ class SaslSslAdminClientIntegrationTest extends AdminClientIntegrationTest with
     assertFutureExceptionTypeEquals(results.results().get(clusterAcl), classOf[InvalidRequestException])
     assertFutureExceptionTypeEquals(results.results().get(emptyResourceNameAcl), classOf[InvalidRequestException])
   }
+
+  val FOO_ACL = new AclBinding(new Resource(ResourceType.TOPIC, "foobar"),
+    new AccessControlEntry("User:ANONYMOUS", "*", AclOperation.READ, AclPermissionType.ALLOW))
+  val FOO_ACL_FILTER = FOO_ACL.toFilter
+
+  private def verifyCauseIsClusterAuth(e: Exception): Unit = {
+    if (!e.getCause.isInstanceOf[ClusterAuthorizationException]) {
+      throw e.getCause
+    }
+  }
+
+  private def testAclCreateGetDelete(expectAuth: Boolean): Unit = {
+    TestUtils.waitUntilTrue(() => {
+      val result = client.createAcls(List(FOO_ACL).asJava, new CreateAclsOptions())
+      if (expectAuth) {
+        Try(result.all().get) match {
+          case Failure(e: Exception) => {
```

Comment:
By adding `e: Exception` you won't match if `e` is a `Throwable`. Better to just not annotate the type at all. Same for other code that does the same.